src/printer.js

```diff
@@ -1,4 +1,46 @@
 // https://prettier.io/docs/en/plugins.html#printers
-const genericPrint = (path, options) => options.originalText;
+const { concat, line } = require('prettier').doc.builders;
+
+const space = ' ';
+
+const genericPrint = (path, options) => {
+  const node = path.getValue();
+  const lines = options.originalText.split('\n');
+  const parts = [];
+
+  switch (node.type) {
+    case 'SourceUnit':
+      node.children.forEach(child => {
+        switch (child.type) {
+          case 'PragmaDirective':
+            parts.push(
+              concat([
+                'pragma',
+                space,
+                child.name,
+                space,
+                child.value,
+                ';',
+                line,
+                line
+              ])
+            );
+            break;
+          default:
+            for (
+              let index = child.loc.start.line;
+              index <= child.loc.end.line;
+              index += 1
+            ) {
+              parts.push(concat([lines[index - 1], line]));
+            }
```

Comment:
I feel like this would be better implemented in this way:

```js
parts.push(options.originalText.slice(child.range[0], child.range[1]))
```

(this requires `range: true` to be passed to the parser)